src/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,494 @@
+	private static final String[][] JSON_INJECTION = {{"$ne", "0"}, {"$gt", ""}, {"$regex", ".*"}};
+	// Error messages that addressing to a well-known vulnerability
+	private  final Pattern[] errorPatterns = {
+			Pattern.compile("RuntimeException: SyntaxError: unterminated string literal", Pattern.CASE_INSENSITIVE),
+			Pattern.compile("MongoResultException", Pattern.CASE_INSENSITIVE)};
+	// Log prints
+	private static final String JSON_EX_LOG = "try to convert the payload in json format";
+	private static final String IO_EX_LOG = "try to send an http message";
+	private static final String URI_EX_LOG = "try to get the message's Uri";
+	private static final Logger LOG = Logger.getLogger(MongoDbInjection.class);
+
+	@Override
+	public int getCweId() {
+		return 943;
+	}
+
+	@Override
+	public int getWascId() {
+		return 19;
+	}
+
+	public int getId() {
+		return 40033;
+	}
+
+	public Tech getTech() {
+		//TODO change in Tech.MongoDB as soon as available
+		return Tech.Db;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {
+		//TODO change in Tech.MongoDB as soon as available
+		return technologies.includes(Tech.Db);
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return new String[] {};
+	}
+
+	public String getExtraInfo(String attack) {
+		return Constant.messages.getString(MESSAGE_PREFIX + "extrainfo." + attack);
+	}
+
+	@Override
+	public void init() {
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Initialising MongoDB penertration tests");
+		}
+		if (this.getAttackStrength() == AttackStrength.LOW) {
+			SLEEP_TIME_SHORT = 3000;
+			SLEEP_TIME_LONG = 5000;
+			doCrashScan = false;
+			doAllDataScan = true;
+			doTimedScan = true;
+			doJsonScan = true;
+			getMoreConfidence = false;
+			doAuthBypass = true;
+		}
+		else if (this.getAttackStrength() == AttackStrength.MEDIUM) {
+			SLEEP_TIME_SHORT = 5000;
+			SLEEP_TIME_LONG = 8000;
+			doCrashScan = true;
+			doAllDataScan = true;
+			doTimedScan = true;
+			doJsonScan = true;
+			getMoreConfidence = true;
+			doAuthBypass = true;
+		}
+		else if (this.getAttackStrength() == AttackStrength.INSANE) {
+			SLEEP_TIME_SHORT = 5000;
+			SLEEP_TIME_LONG = 8000;
+			doCrashScan = true;
+			doAllDataScan = true;
+			doTimedScan = true;
+			doJsonScan = true;
+			getMoreConfidence = true;
+			doAuthBypass = true;
+		}
+	}
+
+	@Override
+	public void scan(HttpMessage msg, NameValuePair originalParam) {
+		//TODO add TYPE_JSON control as soon as available

```

Comment:
I assume this refers to a core change, but I honestly can't recall what it's referring to. Could you please make this more specific or add a reference link?